tools/swift-api-digester/swift-api-digester.cpp

```diff
@@ -1106,6 +1100,12 @@ case SDKNodeKind::X:                                                           \
 // Recursively construct a node that represents a type, for instance,
 // representing the return value type of a function decl.
 static SDKNode *constructTypeNode(SDKContext &Ctx, Type T) {
+  // For generic function type, build from its canonical version.
```

Comment:
Imagine if I have

```
typealias F = Int
var x: F
```

And I change it to
```
var x: Int
```

The type of x is not a GenericFunctionType, it's just a NameAliasType. In this case your tool will still flag it as a source incompatible change. This is why I think it should dump both the sugared and canonical types, or have it be controlled by a flag, instead of just special-casing GenericFunctionType.

In this case canonicalizing a GenericFunctionType is unfriendly for the migrator as well, since you will always lose type alias types in the method type (isCanonical() will always return false, if the function type contains aliases).